### ISSUE TITLE
Little clarification

### ISSUE THREAD

**Issue by nile649** [TalSchuster/VitaminC#1]
So I am a little confused, 

So it seems you're using everything from hugging face, and I am not familiar with all the arguments,

1. Are you training the model first and then evaluating it? https://github.com/TalSchuster/VitaminC#fact-verification , here It says to evaluate?

On running it seems the model spends 20-30min on GPU, so is it training the model and then evaluating? 

I am confused how is training and evaluation being done.

**Comment by TalSchuster** [comment]
Hi @nile649,

Thanks for reaching out. The command in this repository will **load our pre-trained model** (`tals/albert-base-vitaminc-fever`) and evaluate it on VitaminC test. Please see  [scripts/run_fact_verification.sh](https://github.com/TalSchuster/VitaminC/blob/main/scripts/run_fact_verification.sh) for the exact setting. Most of the arguments are imported from HuggingFace and several options were added by us (I'll add some explanations to the README soon).

Actually, the training arguments there don't change anything if you run it as is. But can be used as an example if you want to train again. 

The 20-30 min could be due to an old GPU and the size of the VitaminC test set which is large. You can also run it on your own test file as explained in the instructions. Also you can play with the `per_device_eval_batch_size` and number of threads for the data loader to optimize it to your machine.

In case you want to **train** an ALBERT-base model on VitaminC (I will add FEVER soon). You can use this command for example (add `fp16` if available):
```
    python scripts/fact_verification.py \
      --model_name_or_path albert-base-v2 \
      --tasks_names vitaminc \
      --do_train \
      --do_eval \
      --eval_all_checkpoints \
      --data_dir data \
      --test_on_best_ckpt \
      --do_test \
      --max_seq_length 256 \
      --per_device_train_batch_size 32 \
      --per_device_eval_batch_size 128 \
      --learning_rate 2e-5 \
      --max_steps 50000 \
      --save_step 10000 \
      --overwrite_cache \
      --output_dir results/vitaminc_albert_base \
      --test_tasks vitaminc_real vitaminc_synthetic
``` 
The training will probably take a few hours

**Comment by nile649** [comment]
One more question,

```
{"unique_id": "5ed4de07c9e77c000848a180_1", "case_id": "5ed4de07c9e77c000848a180", "wiki_revision_id": "927477259", "label": "NOT ENOUGH INFO", "claim": "Westlife made under 23.5 million sales in the UK .", "evidence": "According to the British Phonographic Industry ( BPI ) , Westlife has been certified for 13 million albums and 9.8�million singles , **with a total of more than 23 million combined sales in the UK** .", "page": "Westlife", "revision_type": "real"}



{"unique_id": "5ed4de07c9e77c000848a180_2", "case_id": "5ed4de07c9e77c000848a180", "wiki_revision_id": "927477259", "label": "REFUTES", "claim": "Westlife made under 23.5 million sales in the UK .", "evidence": "According to the British Phonographic Industry ( BPI ) , Westlife has been certified for 13 million albums , 1.3 million video albums , and 9.8�million singles , **with a total of more than 24 million combined sales in the UK** .", "page": "Westlife", "revision_type": "real"}
```

These are examples from the vitamin C test file.  Is your model able to understand the difference between numbers?

Reading the first evidence, the model can ignore everything leaving the last few words, and so does the same for the second example. Do you have any insight into how the model is able to see the difference? I have read few papers where the specific task was to understand basic mathematics, but this is different.

**Comment by nile649** [comment]
Also could you explain the reason for displaying input_ids, attention_mask, token_type_ids, and how and what to infer from this?

**Comment by TalSchuster** [comment]
Hi,

The VitaminC data includes many numerical examples among others. Indeed, our models manage to predict correctly on most of these cases, identifying the relevant spans from the sentence and comparing the quantities. 

You can add the `--do_predict` argument to the command line and it will output a file to the output dir with all the model's predictions and examine success/ failure cases.

Regarding the specific two examples above. Our models seem to succeed on the second one but to mistake on the first one and predict it as Support. This is a challenging example for neural models as the claim says "under 23.5" and the evidence has "more than 23". Again, our ALBERT-xlarge baseline model is correct on 89% of the examples in the VitaminC real data, so the vast majority of cases were solved right, but hopefully future models will do even better.

Regarding the input_ids etc. - it just presents the result of the ALBERT tokenization on the input. Could be useful for debugging the tokenizer if needed.

**Comment by nile649** [comment]
Cool, got It. Thanks for the clarification.

**Comment by nile649** [comment]
Question 1 : I have a question while testing your dataset. Is it possible to pass just a claim and evidence to get the result? I want to validate some hypotheses. 
example : 
{"unique_id": "3", "case_id": "3", "wiki_revision_id": "3", "claim": "the price is more than 22 $ a month for the apple plus subscription .", "evidence": "apple plus costs 25 $ a month.", "page": "lol", "revision_type": "real"}



`03/31/2021 15:12:59 - INFO - vitaminc.processing.utils -   features: InputFeatures(input_ids=[2, 213, 3123, 4227, 771, 5579, 21, 1617, 9, 3, 14, 2162, 25, 91, 119, 1024, 5579, 21, 1617, 26, 14, 213, 3123, 14213, 13, 9, 3, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], attention_mask=[1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], token_type_ids=[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], label=1)`


Question 2 : I assumed "label" is the prediction but apparently it is not, so where is the label prediction in the above output.
 


Thank you
